judger/src/tester/runner.rs

```diff
@@ -248,48 +295,53 @@ impl DockerCommandRunner {
                     }
                 };
                 let task = tokio::spawn(task);
-                res.instance
-                    .upload_to_container(
-                        &container_name,
-                        Some(UploadToContainerOptions {
-                            path: to_path.clone(),
-                            ..Default::default()
-                        }),
-                        hyper::Body::wrap_stream(frame.map(|x| x)),
-                    )
-                    .await?;
-                task.await??;
+                try_or_kill!(
+                    r.instance
+                        .upload_to_container(
+                            &container_name,
+                            Some(UploadToContainerOptions {
+                                path: to_path.clone(),
+                                ..Default::default()
+                            }),
+                            hyper::Body::wrap_stream(frame.map(|x| x)),
+                        )
+                        .await
+                );
+                try_or_kill!(try_or_kill!(task.await));
             }
 
-            res.instance
-                .commit_container(
-                    bollard::image::CommitContainerOptions {
-                        container: container_name.clone(),
-                        repo: after_copy_image_name.clone(),
-                        ..Default::default()
-                    },
-                    bollard::container::Config::<String>::default(),
-                )
-                .await?;
+            try_or_kill!(
+                r.instance
+                    .commit_container(
+                        bollard::image::CommitContainerOptions {
+                            container: container_name.clone(),
+                            repo: after_copy_image_name.clone(),
+                            ..Default::default()
+                        },
+                        bollard::container::Config::<String>::default(),
+                    )
+                    .await
+            );
 
-            res.intermediate_images.push(after_copy_image_name.clone());
```

Comment:
Since we will use an external image cache manager, we can remove the tracking of intermediate images.

Reply:
It's generally better to make it optional instead of deleting the feature :)